### ISSUE TITLE
Run architect as a console command

### ISSUE THREAD

**Issue by noandrea** [Pennebaker/craft-architect#6]
It will be very convenient to be able to import and export content models via command line, 
so it can be executed automatically. 

Use case: a staging server could apply changes to the content model.

it should be possible to specify a target path for the export and a source path for the import, for example:
- `./craft architect/export path/to/file.json`
- `./craft architect/import path/to/file.json`

**Comment by timknight** [comment]
👍 

Particularly love this idea for automating imports for project scaffolding.

**Comment by spAnser** [comment]
On the topic of running from command line. I will list my plans on migrations which is what I am working on next. My plan is to implement migrations in a way that uses the built in craft migration system. Which works from command line.

So for example to save a field in craft you have to have a Field class object that is an actual field. You pass this field into a `saveField` function. My plan is to keep track of everything in serialized form. Then from the migrations tab in Architect you can see what has changed since the last migration. If there is changes you can output a migration file that craft can use to make the necessary changes. Since I watch the data before and after I can also create the undo part of the migration.

With that said this is all a theory right now. I may or may not fully use this serialized form to handle the migrations but it will help it spotting changes and hopefully implementing a way to create migrations.

The only issue I can currently see being a problem with this method is migrations to prevent breakage need to keep the same IDs so if two people both created a new field on their own databases they would most likely share the ID which was next in the list. My original plan was to maintain the ID that was used when created locally but that could be problematic if multiple people are working on their own sections. This migrations is a fairly large task and I need to think through all the possibilities that could go wrong. Maintaining IDs is important in the event a field's name or handle is changed making it difficult to determine the matching field on the destination database.

tl;dr

I have a lot to think on when implementing migrations and I would like scenarios or workflows that people might use that could see issues where multiple developers might be adding fields / sections to their local setup that get pushed into a singular place.
